### ISSUE TITLE
Applications not working after modifying config.plist // hibernate crash

### ISSUE THREAD

**Issue by furrybanana** [syscl/XPS9350-macOS#49]
Hey there,

yesterday I had a bad hibernate crash after closing the lid on my XPS 9350.

What I did was modifying my config.plist so I could get iMessage and Facetime to work. So I was generating a Serial, added hex digits, adding uuid etc. and rebooted. Was browsing the web and everything seemed fine. 
I went away and closed the lid, the laptop did go to standby and I opened the laptop a few hours later with a blackscreen but lid keyboard. Nothing happened for about 30 seconds and it did reboot to Clover. After this I wanted to check the drive with diskutil but noticed some strange things on other apps: I couldn't run any of them. I fixed this by reinstalling macOS. 

a screenshot of the apps not working

![error](https://cloud.githubusercontent.com/assets/25349617/22691876/b5f5e070-ed3c-11e6-91e4-2970983aa797.jpg)

**Comment by syscl** [comment]
Hi @furrybanana 

I am sorry to see you encounter this issue. But this is a known issue on XPS 13 and XPS 15 (Skylake). 

Thus I rarely sleep my XPS 13 these days. Could you please provide what you do then see this information?? Because your detail information will help us solve this tough problem!

Thank you,
syscl

**Comment by R1MnTnA** [comment]
hibernatemode is set off so the system will not back memory up to persistent storage. The system must wake from the contents of memory; the system will lose context on power loss or restart.

**Comment by syscl** [comment]
@R1MnTnA But we don't know if hibernate = 0 help to prevent data corruption in this case.

Still need more evidence.

syscl

**Comment by furrybanana** [comment]
I close the lid like any laptop user would do because its a main feature if you have to transport your device. I have a 3-4% battery drain for a 10 hours of hibernate "overnight".
Because I reinstalled I did not yet get this error now, when I encounter this again I hope to get more details. As of now I do a daily CCC-Backup so I don't lose any data.

**Comment by R1MnTnA** [comment]
@syscl No, I was just referring to that hibernate 0 for the fact that he rebooted and got thise pop-ups that he took screenshots of. Also remember the one that I had referenced before in a discussion that we need to investigate further if it helps was not hibernate but the 'pmset -a disksleep 0'

**Comment by R1MnTnA** [comment]
Also just for reference these are what I've done on my MacBook. -
sudo pmset -a hibernatemode 0
sudo pmset -a disksleep 0
sudo pmset -a lidwake 0
sudo rm /var/vm/sleepimage

**Comment by syscl** [comment]
Thank you for the remind.

I will try later. Recently, I am refining rc.script in Clover.

syscl

**Comment by syscl** [comment]
Hello @R1MnTnA 

I want to ask a question, why set lid wake to 0? 

Thank you,
syscl

**Comment by syscl** [comment]
I will try to add this

```sh
sudo rm /private/var/vm/sleepimage
sudo touch /private/var/vm/sleepimage
sudo chflags uchg /private/var/vm/sleepimage
```

syscl

**Comment by R1MnTnA** [comment]
@syscl for me it's only for my preference, not because it is necessary.

**Comment by bozma88** [comment]
Please have a look at my post in issue #26.
The problem is 100% hibernatemode.
If you make some changes like:
- enabling/disabling hwp
- adding descriptors to smbios that trigger a new PM plist to be loaded
- OS update

Your PM settings get reset and hibernatemode reverts to 3 without you noticing.

I had also locked yhe sleepimage but after playing with HWP and frequencyvectors I found the sleepimage REMOVED and hibernwtemode to 3.
After that, I got immediate corruption.

Maybe we can force hibernatemode to 0 before entering sleep, each and every time, with a script.

Since many minor changes can trigger hibernatemode 3, periodic manual checking is too dangerous.

**Comment by ZombieTheBest** [comment]
@bozma88 Deploy script installs Sleepwatcher, which (if i'm not wrong) executes a script when sleep is triggered, and another one when laptop wakes up. maybe we can add in the "sleep script" `sudo pmset -a hibernatemode 0`.

Maybe i will look into this this evening!

**Comment by bozma88** [comment]
We have to be sure that the script sets hibernatemode *before* the system reads the value and decides what to do while sleeping.
I'd also disable darkwake, to be sure, as I'm not sure whether the OS honours the darkwake=0 boot flag.

**Comment by ZombieTheBest** [comment]
@bozma88 we Can check that easily. Lets put hibernatemode 25, edit sleepwatcher script and see if laptop goes to standby or if It hibernates.

**Comment by ZombieTheBest** [comment]
@bozma88 @syscl I got it working. As reported by @bozma88, sometimes hibernatemode value is changed by the OS. We can easily fix this by editing the current `/etc/sysclusbfix.sleep` adding `pmset -a hibernatemode 0` (without sudo) as first line after comments. I tested it and it's working: hibernatemode value is changed before that the OS reads the value. 

Of course it's a temporary solution. Don't forget, after editing the file, to set proper permission with:
```
sudo touch /etc/sysclusbfix.sleep 
sudo chmod +x /etc/sysclusbfix.sleep
```
